### PR TITLE
Update pattern for adding a node

### PR DESCRIPTION
**Pull request by ghost**
This PR:
- changed the update/render function to handle adding a new node
- changed the route to add a text memory so that it didn't refresh after a set time
- put the call to this route in an ajax call on the front end
- re-rendered the new data with a set timeout to wait for the animation to finish
- tweaked the force simulation so that it stopped breaking.
relates #69 #140 #142

### CODE REVIEW

File: src/client/d3/d3.js [hunk 1]
```diff
@@ -218,20 +224,38 @@ function render(updatedData) {
...
           render(formatData(data));
-          sim.restart();
         });
       }
       $.ajax({
         method: 'DELETE',
         url: 'memories',
         data: { id },
-        success: () => update(url),
+        success: () => update(),
```
Comment (des-des) [inline src/client/d3/d3.js]: `success: update`

File: src/client/d3/d3.js [hunk 2]
```diff
@@ -218,20 +224,38 @@ function render(updatedData) {
...
     hideDeleteButton();
-    sim.restart();
+  }
+
+  $('#memory-input__submit').click((e) => {
+    e.preventDefault();
+    $.ajax({
+      method: 'POST',
+      url: 'memory-input-text',
+      data: $('#add-text-form').serialize(),
```
Comment (des-des) [inline src/client/d3/d3.js:242]: Hmm, could you turn this into JSON?

File: src/client/d3/d3.js [hunk 3]
```diff
@@ -218,20 +224,38 @@ function render(updatedData) {
...
+      success: () => {
+        setTimeout(() => {
+          update();
+        }, 4500);
```
Comment (des-des) [inline src/client/d3/d3.js]: ...

This seems like a hack

**Review by des-des — CHANGES_REQUESTED** [review]

File: src/client/d3/d3.js (hunk 3 above)
Comment (ghost) [reply src/client/d3/d3.js]: I don't think it's a hack necessarily... when we add a node, we have a 4.5 second animation so I wanted to wait for that to finish before re rendering the map

File: src/client/d3/d3.js (hunk 2 above)
Comment (ghost) [reply src/client/d3/d3.js:242]: Maybe, but what's the benefit of turning this into JSON?

Comment (des-des) [reply src/client/d3/d3.js:242]: fair

File: src/client/d3/d3.js (hunk 3 above)
Comment (des-des) [reply src/client/d3/d3.js]: Hmm, okay put the 4500 into a variable, you may make it by adding some other variables together

**Review by des-des — APPROVED** [review]

**Pull request merged**